### ISSUE TITLE
mean(f, pointmass) has confusing behaviour for matrix-variate distributions

### ISSUE THREAD

**Issue by bvdmitri** [biaslab/ReactiveMP.jl#324]
Currently, the `mean(f, pointmass)` function exhibits confusing behavior for `Matrixvariate` distributions. Please consider the following code from ReactiveMP:

```julia
mean(::typeof(cholinv), distribution::PointMass{M}) where {T <: Real, M <: AbstractMatrix{T}}   = cholinv(mean(distribution))
mean(::typeof(log), distribution::PointMass{M}) where {T <: Real, M <: AbstractMatrix{T}}       = log.(mean(distribution))
```

<p>Why does <code>mean(log, pointmass)</code> take the point-wise logarithm in the case of a Matrix? I suppose the answer here is that it's convenient, but it is incorrect from a mathematical perspective. The <code>mean(function, distributions)</code> computes an expectation of the <code>distribution</code> with respect to the <code>function</code>, where <code>mean(distribution)</code> is technically defined as <code>mean(x -&gt; x, distribution)</code>. Therefore, <code>mean(log, A)</code> should compute the expectation of the principal matrix logarithm of A instead of the point-wise logarithm of A.</p>

Perhaps, we should define something like instead?
```julia
mean(f, PointWise(), pointmass)
```
<p>Or should we disallow this behavior and require the user to write <code>f.(mean(pointmass))</code> instead? I believe we are using <code>mean(log, distribution)</code> somewhere, and it could be incorrect for matrix-variate pointmass distributions. It might be best to double-check this.</p>

What do you think, @albertpod @ThijsvdLaar @bartvanerp @ismailsenoz?

This issue was initially encountered by @ThijsvdLaar

**Comment by bvdmitri** [comment]
Or maybe we do `mean(PointWise(f), distribution)` to avoid any confusion.

**Comment by bartvanerp** [comment]
> Therefore, `mean(log, A)` should compute the expectation of the principal matrix logarithm of A instead of the point-wise logarithm of A.

I fully agree here. It should be mathematically correct. The current behavior could lead to unexpected bugs. This issue is also the case for `AbstractVector`.

> mean(f, PointWise(), pointmass)

Would `mean(f., pointsmass)` also work? Otherwise I think it is best to stick to Julia terminology and use `mean(f, Broadcast(), pointmass)`.

**Comment by bvdmitri** [comment]
I like the `mean(f., pointmass)` indeed.

**Comment by bvdmitri** [comment]
But it does not parse properly :(

**Comment by bvdmitri** [comment]
Should be addressed by https://github.com/biaslab/BayesBase.jl/tree/dev-1.0

**Comment by bvdmitri** [comment]
Fixed in #362
